rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -2309,6 +2310,78 @@ void getProposedFederationCreationBlockNumber_whenStorageProviderReturnsProposed
         assertEquals(creationBlockNumber, actualCreationBlockNumber.get());
     }
 
+    @ParameterizedTest
+    @EnumSource(FederationMember.KeyType.class)
+    void getProposedFederatorPublicKeyOfType_whenFederationIsEmpty_shouldReturnEmpty(FederationMember.KeyType keyType) {
+        // Act
+        Optional<byte[]> actualPublicKey = federationSupport.getProposedFederatorPublicKeyOfType(0, keyType);
+
+        // Assert
+        assertFalse(actualPublicKey.isPresent());
+    }
+
+    @Test
+    void getProposedFederatorPublicKeyOfType_whenStorageProviderReturnsProposedFederationAndKeyTypeIsBTC_shouldReturnPublicKey() {
+        // Arrange
+        List<BtcECKey> federationKeys = BitcoinTestUtils.getBtcEcKeysFromSeeds(
+            new String[] { "fa01", "fa02", "fa03", "fa04", "fa05", "fa06", "fa07", "fa08", "fa09" }, true);
+        Federation proposedFederation = P2shErpFederationBuilder.builder().withMembersBtcPublicKeys(federationKeys).build();
+        FederationMember.KeyType keyType = FederationMember.KeyType.BTC;
+        byte[] expectedPublicKey = proposedFederation.getMembers().get(0).getPublicKey(keyType).getPubKey(true);
```

Comment:
```suggestion
byte[] expectedPublicKey = federationKeys.get(0).getPubKey();
```
according to the discussion we had, right?

Reply:
The order might be different, first key in `federationKeys` might not match with the first member. Keys are sorted by hex value